src/AppInstallerCLICore/Workflows/ArchiveFlow.cpp

```diff
@@ -32,28 +35,56 @@ namespace AppInstaller::CLI::Workflow
         const auto& installer = context.Get<Execution::Data::Installer>().value();
         if (installer.NestedInstallerFiles.empty())
         {
-            // Manifest validation should prevent this from happening
+            // Pre-install validation should prevent this from happening
             AICLI_LOG(CLI, Error, << "No entries specified for NestedInstallerFiles");
             AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_INVALID_MANIFEST);
         }
 
         const auto& installerPath = context.Get<Execution::Data::InstallerPath>();
         const auto& installerParentPath = installerPath.parent_path();
-
         const auto& relativeFilePath = ConvertToUTF16(installer.NestedInstallerFiles[0].RelativeFilePath);
 
-        std::filesystem::path nestedInstallerPath = installerParentPath / relativeFilePath;
+        const std::filesystem::path& nestedInstallerPath = installerParentPath / relativeFilePath;
 
         if (!std::filesystem::exists(nestedInstallerPath))
         {
             AICLI_LOG(CLI, Error, << "Unable to locate nested installer at: " << nestedInstallerPath);
             context.Reporter.Error() << Resource::String::NestedInstallerNotFound << ' ' << nestedInstallerPath << std::endl;
             AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_NESTEDINSTALLER_NOT_FOUND);
         }
+        else if (Filesystem::PathEscapesBaseDirectory(nestedInstallerPath, installerParentPath))
```

Comment:
The manifest validation should also do this kind of check to doubly prevent it from happening (and to prevent a manifest with such a thing from getting into `winget-pkgs`).

Reply:
Added a manifest validation check for checking the relative path using the std::current_path as the base path. I also rearranged the order of the checks above so that verifying the relativeFilePath comes before checking if the nestedInstaller exists as that logically makes more sense.